pkg/network/protocols/http/ebpf_javatls.go

```diff
@@ -78,7 +106,40 @@ func (p *JavaTLSProgram) GetAllUndefinedProbes() (probeList []manager.ProbeIdent
 	return
 }
 
+// isAttachmentAllowed will return true if the pid must not be attach
+// The filter is based on the process command line matching javaAgentAllowRegex and javaAgentBlockRegex regex
+// javaAgentAllowRegex have an higher priority
+func isAttachmentAllowed(pid uint32) bool {
+	// filter is disabled (default configuration)
+	if javaAgentAllowRegex == nil && javaAgentBlockRegex == nil {
+		return true
+	}
+
+	procCmdline := fmt.Sprintf("%s/%d/cmdline", util.HostProc(), pid)
+	cmd, err := os.ReadFile(procCmdline)
+	if err != nil {
+		log.Debugf("injectionFilter can't open comandline %s : %s", procCmdline, err)
+		return false
+	}
+	fullCmdline := strings.ReplaceAll(string(cmd), "\000", " ") // /proc/pid/cmdline format : arguments are separated by '\0'
+
+	// Allow have an higher priority
+	if javaAgentAllowRegex != nil && javaAgentAllowRegex.MatchString(fullCmdline) {
+		return true
+	}
+	if javaAgentBlockRegex != nil && javaAgentBlockRegex.MatchString(fullCmdline) {
+		return false
+	}
+	// by default we attach the java process
+	return true
+}
```

Comment:
let's add UTs to that function
consider the following scenario:
``` go
javaAgentBlockRegex = nil
javaAgentBlockRegex = "myProg.*"
```

Now a process called "my prog" is being spawned, we should not attach to that process, but with the current implementation we will

Reply:
```
javaAgentAllowRegex = nil
javaAgentBlockRegex = "myProg.*"
```
"myProg.*" regex will not match "my prog" command line so we fallback to default return true

Could you elaborate your point ?